src/main/kotlin/lotto/domain/Prize.kt

```diff
@@ -3,16 +3,20 @@ package lotto.domain
 enum class Prize(
     val value: Int,
     val matchCount: Int,
-    private val matcher: (Int) -> Boolean,
+    private val matcher: (Int, Boolean) -> Boolean,
 ) {
-    FIRST(2_000_000_000, 6, { it == 6 }),
-    SECOND(1_500_000, 5, { it == 5 }),
-    THIRD(50_000, 4, { it == 4 }),
-    FOURTH(5_000, 3, { it == 3 }),
-    NONE(0, 0, { false }),
+    FIRST(2_000_000_000, 6, { count, _ -> count == 6 }),
+    SECOND(30_000_000, 5, { count, hasBonus -> count == 5 && hasBonus }),
+    THIRD(1_500_000, 5, { count, hasBonus -> count == 5 && !hasBonus }),
+    FOURTH(50_000, 4, { count, _ -> count == 4 }),
+    FIFTH(5_000, 3, { count, _ -> count == 3 }),
+    NONE(0, 0, { _, _ -> false }),
```

Comment:
Adding bonus logic has made the conditional statement more complicated. 🤔 

Especially the 2nd and 3rd places don't need to know if they have bonuses, so it might feel unnecessary to be included as a property on enum. 

How do you think Sun?

Reply:
I was thinking that the `hasBonus` is also part of the prize domain because it plays a significant role in differentiating SECOND and THIRD prize. However doing so is making the code more complicated with repetitions or unnecessary underscores. Even if I added a property with `val requireBonus`, the finding logic seems more difficult. 

As a result, I refactored so that the prize is calculated on the `calculate` method. I feel like this has made the logic more simpler. Hope this was what you meant. 🙌